### DISCUSSION TITLE
codeql vscode plugin show result in `#select` page but not `alert` page

### DISCUSSION THREAD

**Discussion by KiruaLawliet** [github/codeql#7505] [Q&A] [answered]
evnironment:
CodeQL command-line toolchain release 2.7.3.
vscode Version: 1.63.2
codeql plugin ver：v1.5.9
OS: Windows_NT x64 10.0.19042

database of project A：compiled from openjdk8u which is a huge java project（about 18,800+ java files）
database of project B：compiled from project A with additional two java projects（rome1.0 and rhino:js:1.7R2）

After successfully compiling , I checked database directory of project B and seen all source files of three projects included in src.zip file.
However，when I run query script individually on each database, query result of porject B is strange.

<h3> result of project A </h3>

this result is exactually what I want.

![image](https://user-images.githubusercontent.com/16208969/147809426-714edf66-2e84-4a75-b10d-eb9a47a33a94.png)

<h3> result of project B </h3>

All path nodes in the result showed above are from porject A.  Thus,  it should come out to share the same part of  query result, 
but as you can see  below，no result in  `alert` page, 

![image](https://user-images.githubusercontent.com/16208969/147809475-8497e6ef-d8e7-44c7-be4f-42d0999e766b.png)

just one result in `#select` page and not in path-problem view

![image](https://user-images.githubusercontent.com/16208969/147809489-114ab3f0-9408-43e9-a7ec-930f38940655.png)

don't know why, here are my simplify query script which got `isAdditionalTaintStep` removed：
```sql

/**@kind path-problem */

import java
import semmle.code.java.dataflow.TaintTracking

import DataFlow::PathGraph

class DeserialFlowConfig extends TaintTracking::Configuration
{
    DeserialFlowConfig()
    {
        this = "DeserialFlowConfig" 
    }

    override predicate isSource(DataFlow::Node node) {
        node.asParameter().getCallable().hasQualifiedName("java.util", "HashMap", "hash")
        and node.asParameter().getPosition() = 0
    }

    override predicate isSink(DataFlow::Node node) {
        exists(MethodAccess ma|
            ma.getMethod().hasName("hashCode")
            and ma.getEnclosingCallable().hasName("hashCode")
            and ma.getAnArgument() = node.asExpr()
            and ma.getMethod().getDeclaringType().hasQualifiedName("java.net", "URLStreamHandler")
        )

    }
}


from DeserialFlowConfig config, DataFlow::PathNode source, DataFlow::PathNode sink
where config.hasFlowPath(source, sink)
select sink.getNode(), source, sink, "RESULT"


```

**Comment by smowton** [comment, marked as answer]
Responded at https://github.com/github/securitylab/discussions/516

**Reply by KiruaLawliet** [reply]
@smowton  Thanks again.
